### PR TITLE
[auth] refactor auth flow

### PR DESCRIPTION
**Pull request by me-liu**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- created an AuthButton, and changed all buttons to that button
- changed some of the wording to match the figma

### Screenshots
- lowkey it looks the same

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ronniebeggs

### CODE REVIEW

File: src/app/(Authentication)/Welcome/index.tsx [hunk 1]
```diff
@@ -15,19 +22,29 @@ function WelcomeScreen() {
...
+            <Text style={styles.orStyle}>OR</Text>
+            <GreyLine style={{ margin: 5 }} />
+          </View>
+          <ButtonBlack
+            style={styles.baseButton}
+            onPress={() => router.push('/SignUp/Email')}
+          >
+            <Text style={styles.nextText}>Log in with email</Text>
+            <RightArrowWhite style={{ marginRight: 10 }} />
+          </ButtonBlack>
```
Comment (ronniebeggs) [inline src/app/(Authentication)/Welcome/index.tsx]: Login button currently routes to the sign up flow.

File: src/app/(Authentication)/OTPFlow/OTPVerify/index.tsx [hunk 2]
```diff
@@ -76,10 +73,11 @@ export default function OTPFlow() {
         />
 
         <Text style={styles.instructionText}>
-          Didn't receive a code? Go back to confirm your email or{' '}
+          Didn't receive a code? Go back to confirm your email or
           <TouchableOpacity onPress={() => resendOtp(email)}>
             <Text style={[styles.instructionText, styles.underlineText]}>
-              press here to resend code.
+              {' '}
+              tap here to resend it.
```
Comment (ronniebeggs) [inline src/app/(Authentication)/OTPFlow/OTPVerify/index.tsx]: Can we underline this text so it's clear where to press.

File: src/app/(Authentication)/OTPFlow/OTPVerify/index.tsx [hunk 3]
```diff
@@ -90,20 +88,16 @@ export default function OTPFlow() {
...
 
-        <TouchableOpacity
+        <ButtonBlack
           disabled={token.length !== 6 || errorExists}
-          style={
-            token.length !== 6 || errorExists
-              ? [styles.nextButtonBase, styles.nextButtonDisabled]
-              : [styles.nextButtonBase, styles.nextButtonActive]
-          }
+          style={styles.nextButtonBase}
```
Comment (ronniebeggs) [inline src/app/(Authentication)/OTPFlow/OTPVerify/index.tsx]: The styled component shouldn't have additional styling unless this instance is different from others.

File: src/app/(Authentication)/OTPFlow/OTPVerify/index.tsx [hunk 4]
```diff
@@ -58,9 +57,7 @@ export default function OTPFlow() {
       <View style={styles.contentContainer}>
         <View style={styles.instructionContainer}>
           <Text style={styles.headerText}>Enter verification code.</Text>
-          <Text style={styles.instructionText}>
-            We've just emailed it to you at {email}
-          </Text>
+          <Text style={styles.instructionText}>We've sent it to {email}</Text>
```
Comment (ronniebeggs) [inline src/app/(Authentication)/OTPFlow/OTPVerify/index.tsx]: Apply a 100% width so it fills the content container and has the same width as the OPT container below.

File: src/app/(Authentication)/SignUp/Address/index.tsx [hunk 5]
```diff
@@ -122,28 +123,21 @@ export default function SignUpScreen() {
...
         }
-        style={
-          streetAddress.trim() === '' ||
-          city.trim() === '' ||
-          state.trim() === '' ||
-          zipcode.trim() === ''
-            ? styles.nextButtonGray
-            : styles.nextButton
-        }
+        style={styles.nextButton}
```
Comment (ronniebeggs) [inline src/app/(Authentication)/SignUp/Address/index.tsx]: Remove this redundant style prop

File: src/app/(Authentication)/SignUp/Address/index.tsx [hunk 6]
```diff
@@ -122,28 +123,21 @@ export default function SignUpScreen() {
...
         onPress={() => handleSubmit()}
       >
         <Text style={styles.nextText}>Sign Up</Text>
         <View style={styles.check}>
           <Check />
         </View>
```
Comment (ronniebeggs) [inline src/app/(Authentication)/SignUp/Address/index.tsx]: can we remove the view that wraps the check icon. Instead wrap both the "Sign Up" text and icon with one View container that centers them along the same axis and separates them by a column gap. That way we can remove margins from text components.

File: src/app/(Authentication)/Welcome/index.tsx [hunk 7]
```diff
@@ -15,19 +22,29 @@ function WelcomeScreen() {
...
+          >
+            <Text style={styles.nextTextBlack}>Create an account</Text>
+            <RightArrow style={{ marginRight: 10 }} />
+          </ButtonWhite>
+
+          <View style={styles.orLine}>
+            <GreyLine style={{ margin: 5 }} />
+            <Text style={styles.orStyle}>OR</Text>
+            <GreyLine style={{ margin: 5 }} />
+          </View>
```
Comment (ronniebeggs) [inline src/app/(Authentication)/Welcome/index.tsx]: I'd prefer to use a container with heigh 0 that has a bottom width. That way it expands and shrinks depending on the screen size, whereas an SVG line has the same dimensions regardless. Check the `/Components/EligibilityCard` to see how this is done.

File: src/app/(Authentication)/Welcome/styles.ts [hunk 8]
```diff
@@ -13,18 +13,15 @@ export default StyleSheet.create({
   },
   contentContainer: {
     height: '84%',
-    width: '82%',
-  },
-  imageContainer: {
-    width: '100%',
+    width: 315,
```
Comment (ronniebeggs) [inline src/app/(Authentication)/Welcome/styles.ts]: Can we keep this as a percentage width.

File: src/Components/AuthButton/AuthButton.tsx [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+`;
+
+export const ButtonBlack = styled(ButtonBase)`
+  background: ${colors.black};
+`;
+
+export const ButtonWhite = styled(ButtonBase)`
+  background: ${colors.white};
+  border-color: ${colors.black};
+  border-width: 2px;
```
Comment (ronniebeggs) [inline src/Components/AuthButton/AuthButton.tsx]: Border width should be 1px

File: src/app/(Authentication)/Login/Email/index.tsx [hunk 10]
```diff
@@ -32,7 +34,7 @@ export default function LoginScreen() {
   return (
     <View style={styles.container}>
       <TouchableOpacity style={styles.backButton} onPress={() => router.back()}>
```
Comment (ronniebeggs) [inline src/app/(Authentication)/Login/Email/index.tsx]: Can we add a `contentContainer` View that defines the width of the content on this screen. That way all components have a consistent width.

**Review by ronniebeggs — CHANGES_REQUESTED** [review]
Looks good so far. A few simplifying changes should be applied across screens to make styling consistent.

**Review by stephaniewong2 — COMMENTED** [review]
remove back button on forgot password flow

**Pull request merged**